Shared/Extensions/WKWebViewExtensions.swift

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import JavaScriptCore
+import WebKit
+
+enum JavascriptError: Error {
+    case invalid
+}
+
+extension WKUserScript {
+    public class func createInDefaultContentWorld(source: String, injectionTime: WKUserScriptInjectionTime, forMainFrameOnly: Bool) -> WKUserScript {
+        if #available(iOS 14.0, *) {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly, in: .defaultClient)
+        } else {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly)
+        }
+    }
+}
+
+public extension WKWebView {
+    func generateJavascriptFunctionString(functionName: String, args: [Any], escapeArgs: Bool = true) -> (javascript: String, error: Error?) {
+        let context = JSContext()
+
+        var sanitizedArgs: [String] = []
+        var error: Error?
+
+        args.forEach {
+            if !escapeArgs {
+                sanitizedArgs.append("\($0)")
+                return
+            }
+            context?.exceptionHandler = { context, exception in
+                if exception != nil {
+                    error = JavascriptError.invalid
+                }
+            }
+            context?.evaluateScript("JSON.parse('\"\($0)\"')")
+            sanitizedArgs.append("'\(String(describing: $0).escapeHTML())'")
```

Comment:
Why bother with `JSContext` if you're not going to use the resulting `JSValue` and still use `escapeHTML`? Shouldn't we be only appending a sanitized arg if it successfully parses?

Reply:
We are doing a two level check:

1. `JSON.parse` exceptions help verify if the object is a valid JS object. Whenever JSON.parse throws an exception a `JavascriptError.invalid` will be thrown.

2. `escapeHTML` is a second level check which sanitizes the inputs before invoking `evaluateJavascript`